sdk/eventhub/Azure.Messaging.EventHubs/assets/functions/resource-management-functions.ps1

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+using namespace Microsoft.Azure.Commands.ActiveDirectory
+
+function TearDownResources 
```

Comment:
The intent of this was to tear down only what the script was directly responsible for creating, in the event of failure.   We really don't want to remove things that were already in place before the script ran, since the script doesn't own them.

We'll need a way to do that.... maybe by having each `CreateIfMissing` function return a flag to indicate that it created something and then having each script implement its version of this.   Maybe using a helper method defined here that has the messaging and takes a script block to execute (or not)?